multiraft/multiraft.go

```diff
@@ -623,22 +623,22 @@ func (s *state) stop() {
 	s.MultiRaft.Transport.Stop(s.nodeID)
 
 	// Drain the input channels because other threads may be blocking
-	// on these operations.
-	for {
-		select {
-		case op := <-s.createGroupChan:
-			op.ch <- util.Errorf("shutting down")
-		case op := <-s.removeGroupChan:
-			op.ch <- util.Errorf("shutting down")
-		case op := <-s.proposalChan:
-			op.ch <- util.Errorf("shutting down")
-		default:
-			close(s.createGroupChan)
-			close(s.removeGroupChan)
-			close(s.proposalChan)
-			return
+	// on these operations. We can't close these channels unless we
+	// make sure that we never write to them after, but that would mean
+	// an atomic guard before every write access. Better to just deal
```

Comment:
On the store side all of these things should already be atomically guarded by `Stopper.StartTask`. 

Leaking a goroutine isn't a big problem since this won't happen more than once per process in production (although this implementation holds a reference to the `state` object and might indirectly keep a bunch of objects from GC), but it just seems clumsy to require it.